src/measurement/handler/request.ts

```diff
@@ -0,0 +1,12 @@
+import type { Probe } from '../../probe/types.js';
+import { getProbeValidator } from '../../lib/probe-validator.js';
+import { MeasurementRequestMessage } from '../types.js';
+
+export const listenMeasurementRequest = (probe: Probe) => (event: string, data: unknown) => {
+	if (event !== 'probe:measurement:request') {
+		return;
+	}
+
+	const message = data as MeasurementRequestMessage;
+	getProbeValidator().addValidIds(message.measurementId, message.testId, probe.uuid);
```

Comment:
Any reason we're adding it indirectly here instead of inside `sendToProbes` ?

Reply:
If called inside `sendToProbes` it will add ids to the process initiating the measurement. While what we need is to add id to the process where particular probe is connected, so every process listens for `probe:measurement:request` event using `socket.onAnyOutgoing` and stores the id in memory before sending message to the probe.